src/s3path.jl

```diff
@@ -1,14 +1,26 @@
-struct S3Path <: AbstractPath
+struct S3Path{A<:AbstractAWSConfig} <: AbstractPath
     segments::Tuple{Vararg{String}}
     root::String
     drive::String
     isdirectory::Bool
-    config::AWSConfig
+    config::A
+
+    # this method ensures that there is proper conversion to Tuple{Vararg{String}} for first arg,
+    # which can be a little tempremental if you don't explicitly handle it
+    function S3Path{A}(segments, root::AbstractString, drive::AbstractString, isdirectory::Bool,
+                       config::A) where {A<:AbstractAWSConfig}
```

Comment:
```suggestion
    function S3Path{A}(
        segments,
        root::AbstractString,
        drive::AbstractString,
        isdirectory::Bool,
        config::A
    ) where {A<:AbstractAWSConfig}
```

Reply:
....GH suggestions just don't seem to be happy. Breaking this into the 2nd multi-line if needed fits better with the style guide

Reply:
That seems rather excessively verbose, but ok, committing.